modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -25,51 +26,24 @@ class AbstractSelectionStrategy(ABC):
     """
 
     # pylint: disable-next=too-many-branches
-    def __init__(
-        self,
-        config: dict,
-        modyn_config: dict,
-        pipeline_id: int,
-        maximum_keys_in_memory: int,
-        required_configs: Optional[list[str]] = None,
-    ):
+    def __init__(self, config: SelectionStrategy, modyn_config: dict, pipeline_id: int):
         self._config = config
 
-        if required_configs is None:
-            required_configs = []  # Using [] as default is considered unsafe by pylint
-
-        required_configs.extend(["limit", "reset_after_trigger"])
-        for required_config in required_configs:
-            if required_config not in self._config.keys():
-                raise ValueError(f"{required_config} not given but required.")
-
-        self.training_set_size_limit: int = config["limit"]
+        self.training_set_size_limit: int = config.limit
         self.has_limit = self.training_set_size_limit > 0
-        self.reset_after_trigger: bool = config["reset_after_trigger"]
 
         # weighted optimization (with weights supplied by the selector) is quite unusual, so the default us false
-        if config.get("uses_weights", None):
-            self.uses_weights = config["uses_weights"]
-        else:
-            self.uses_weights = False
-
-        if config.get("tail_triggers", None):
-            self.tail_triggers = config["tail_triggers"]
-            if self.tail_triggers < 0 or not isinstance(config["tail_triggers"], int):
-                raise ValueError("Tail trigger must be an integer greater than 0")
-            if (self.reset_after_trigger and self.tail_triggers > 0) or (
-                (not self.reset_after_trigger) and self.tail_triggers == 0
-            ):
-                raise ValueError("Reset after trigger is equivalent to setting tail triggers to 0.")
+        self.uses_weights = config.uses_weights
+
+        self.tail_triggers = config.tail_triggers
```

Comment:
What if `tail_triggers` is not given at all? Wouldn't this lead to a key error? do we need something like get(tail_triggers, None)?

Reply:
Even previously, `tail_triggers` is only optionally specified by the user, but will be a field in the Pydantic model absolutely (when value `None` if user does not specify it).

I also removed the default value provided by Pydantic if user does not specify it, because I think it's better to always let user explicitly specify it than relying on an unobvious implicit default.

An obvious implicit default is, for example, `NoPresamplingStrategy`

Reply:
Mind that the type annotation in pydantic only defines what the user of the model can expect. For a missing field with a model specification of `field: int | None` pydantic will raise missing value errors only if no default (e.g. None) is specified.

Reply:
ahh, I see. sorry i might have been a bit confused. thanks